### PR TITLE
Add a comment to explain the struct pattern trick.

### PR DESCRIPTION
**Pull request by SimonSapin**
https://github.com/servo/servo/issues/6912#issuecomment-127429643

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6950)

### CODE REVIEW

File: components/util/mem.rs [hunk 1]
```diff
@@ -103,6 +103,8 @@ impl<T: HeapSizeOf> HeapSizeOf for Option<T> {
 
 impl HeapSizeOf for url::Url {
     fn heap_size_of_children(&self) -> usize {
+        // Using a struct pattern without `..` rather than `foo.bar` field access
+        // makes sure this will be updated in a field is added.
```
Comment (glennw) [inline components/util/mem.rs]: in -> if

File: components/util/mem.rs [hunk 2]
```diff
@@ -122,6 +124,8 @@ impl HeapSizeOf for url::SchemeData {
 
 impl HeapSizeOf for url::RelativeSchemeData {
     fn heap_size_of_children(&self) -> usize {
+        // Using a struct pattern without `..` rather than `foo.bar` field access
+        // makes sure this will be updated in a field is added.
```
Comment (glennw) [inline components/util/mem.rs]: in -> if

**Comment by jdm** [comment]
@bors-servo: r-
Fight!

**Comment by SimonSapin** [comment]
Typos fixed.

**Pull request merged**